packages/calcite-components/src/components/combobox/combobox.tsx

```diff
@@ -686,14 +713,37 @@ export class Combobox
     this.open = false;
   }
 
+  private handleSelectAll(isSelectAllTarget: boolean): void {
+    if (isSelectAllTarget) {
+      this.toggleSelectAll();
+    }
+
+    if (this.allSelected) {
+      this.selectedItems.forEach((item) => {
+        const chipEl = this.referenceEl.querySelector<Chip["el"]>(`#${chipUidPrefix}${item.guid}`);
+        if (chipEl) {
+          this.hideChip(chipEl);
+        }
+      });
+    }
+  }
+
   private calciteComboboxItemChangeHandler(
     event: CustomEvent<HTMLCalciteComboboxItemElement["el"]>,
   ): void {
     if (this.ignoreSelectedEventsFlag) {
       return;
     }
-
     const target = event.target as HTMLCalciteComboboxItemElement["el"];
+
+    const isSelectAllTarget = event
+      .composedPath()
+      .some((node) => (node as HTMLElement) === this.selectAllComboboxItemReferenceEl);
```

Comment:
nitpick: I think elsewhere in the codebase we use `.composedPath().includes(this.selectAllComboboxItemReferenceEl)` which I think is more performant.